src/components/ui/datepicker.tsx

```diff
@@ -0,0 +1,161 @@
+"use client";
+
+import * as React from "react";
+import { format, getMonth, getYear, setMonth, setYear } from "date-fns";
+import { Calendar as CalendarIcon } from "lucide-react";
+import { cn } from "@/lib/utils";
+import { Button } from "@/components/ui/button";
+import { Calendar } from "@/components/ui/calendar";
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+} from "@/components/ui/popover";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "./select";
+import {
+  FormField,
+  FormItem,
+  FormLabel,
+  FormControl,
+  FormMessage,
+} from "./form";
+import { Control, FieldPath, FieldValues } from "react-hook-form";
+
+interface DatePickerProps<T extends FieldValues> {
+  startYear?: number;
+  endYear?: number;
+  label: string;
+  control: Control<T>;
+  name: FieldPath<T>;
+}
+
+export function DatePicker<T extends FieldValues>({
+  startYear = getYear(new Date()) - 60,
```

Comment:
i think you no need this prop `startyear` and `endYear` you can direclty use in this component

Reply:
Okay Fixed